core/unit_test/TestGraph.hpp

```diff
@@ -127,7 +127,50 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), launch_one_rvalue) {
   ASSERT_EQ(0, bugs_host());
 }
 
-TEST_F(TEST_CATEGORY_FIXTURE(graph), launch_six) {
+/// @test Ensure that @c Kokkos::Graph::instantiate works.
+/// @note For now, @c Kokkos::Graph::submit will instantiate if needed,
+/// so this test is not very strong.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), instantiate_and_submit_once) {
+  auto graph = Kokkos::Experimental::create_graph(ex, [&](auto root) {
+    root.then_parallel_for(1, count_functor{count, bugs, 0, 0});
+  });
+  graph.instantiate();
+  graph.submit();
+  Kokkos::deep_copy(ex, count_host, count);
+  Kokkos::deep_copy(ex, bugs_host, bugs);
+  ex.fence();
+  ASSERT_EQ(1, count_host());
+  ASSERT_EQ(0, bugs_host());
+}
+
+/// @test Ensure that @c Kokkos::Graph::instantiate can be called only once.
+/// This test checks 2 cases:
+///   1. Instantiating after submission is invalid (this also implicitly checks that submission instantiates if need be).
+///   2. Instantiating twice in a row is invalid.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), can_instantiate_only_once) {
+#if !defined(NDEBUG) || defined(KOKKOS_ENABLE_DEBUG)
```

Comment:
Actually I can think of a cute way of checking if assertions are checked:
```C++
bool checked_assertions = false;
KOKKOS_ASSERT(checked_assertions = true);
if (!checked_assertions) {
  GTEST_SKIP() << "Preconditions are not checked.";
}
```
(not tested)

Reply:
OK, done.

I came with a macro. Its name could be better (`GTEST_SKIP_IF_NO_ASSERTION`), but at least it keeps the test body focused on the test, and makes this logic reusable.

Reply:
We will make it reusable when we need to reuse it

Reply:
You're right. I quickly came over the other test files and it seems it cannot be reused for now. I'll revert.